restaurant-menu/src/containers/CustomerInfo/CustomerInfo.js

```diff
@@ -0,0 +1,360 @@
+            newValidation[fieldName] = true;
+            if (event.target.value === null || event.target.value === "")
+                newValidation[fieldName] = false;
+        }
+
+        if (["city", "street"].includes(fieldName)){
+            newOrder[fieldName] = event.target.value;
+            newValidation[fieldName] = true;
+            if (event.target.value === null || event.target.value === "")
+                newValidation[fieldName] = false;
+        }
+
+        if (fieldName === "number"){
+            newOrder.number = event.target.value;
+            newValidation.number = false;
+            if (event.target.value > 0 )
+                newValidation.number = true;
+        }
+
+        if (fieldName === "apartment"){
+            newOrder.apartment = event.target.value;
+            newValidation.apartment = true;
+
+            if (event.target.value <= 0 && !(event.target.value === null || event.target.value === ""))
+                newValidation.apartment = false;
+        }
+
+        if (fieldName === "remark")
+            newOrder.remark = event.target.value;
+
+        setOrder(newOrder);
+        setValidation(newValidation);
+    }
+
+    const transformOrderToIds = order => {
+        order.dishes_in_order.forEach(dishInOrder => {
+            dishInOrder.dish = dishInOrder.dish.id;
+            for(let i=0; i < dishInOrder.extras.length; i++) {
+                dishInOrder.extras[i] = dishInOrder.extras[i].id;
+            }
+        });
+        return order;
+    }
+
+    const csrftoken = getCookie('csrftoken');
+    const originURL = window.location.origin;
+
+    const submitHandler = async event => {
+        
+        event.preventDefault();
+        let valid;
+        const newOrder = {...order};
+        try{
+            const geoData = await allValid();
+            newOrder.address_lat = geoData.customerLat;
+            newOrder.address_lng = geoData.customerLng;
+            newOrder.delivery_distance = geoData.deliveryDistance;
+            valid = true;
+        }
+        catch (value){
+            valid = false;
+        }
+        if (valid){
+            setStatus("loading");
+            const orderWithIds = transformOrderToIds(newOrder);
+            axios.post(originURL + '/order_add', orderWithIds,
+                {
+                    headers: {'X-CSRFTOKEN': csrftoken,},
+                },
+            )
+            .then(res => {
+                console.log("SUCCESS adding order")
+                props.setPage({page: "success", info: res.data.id})
+            })
+            .catch(err => {
+                console.log("ERROR posting order");
+                console.log(err.message);
+                setStatus("error");
+            })
+        }
+        else
+            setSubmitValidation(validation);  
+    }
+
+    let content = null;
+    if (!status) {
+        content = (
+            <React.Fragment>
```

Comment:
This component is way too long, this jsx part of way too long, should separated to multiple components